### ISSUE TITLE
theme(text = ... ) not affecting axis text

### ISSUE THREAD

**Issue by rjake** [tidyverse/ggplot2#4909]
I wish I had seen `theme(text = ... )` earlier. It's so nice! I am expecting these aesthetics to flow through to `axis.text`. The axis titles pick it up but not the axis text.

```r
library(tidyverse)

quickplot(mpg, wt, data = mtcars) + 
  theme(text = element_text(colour = "red"))
```

![image](https://user-images.githubusercontent.com/14003618/179126579-d0d6dff4-768a-42ef-9a3d-35b56927f809.png)

After watching [Kara Woo's talk](https://resources.rstudio.com/resources/rstudioglobal-2021/always-look-on-the-bright-side-of-plots/) I think it's because the color is hard coded in `theme_grey()` but should it be?
https://github.com/tidyverse/ggplot2/blob/a979ffd26cdb456d54e2671c2eed16c65bc878b7/R/theme-defaults.r#L137

I thought all of these would update:
<details>

```r
# by name
theme_grey() |> 
  names() |> 
  keep(str_detect, pattern = "(caption|tag|text|title)(\\.[xy].*)?$") |>
  glue::glue_collapse("\n")

# by class
theme_grey() |> 
  map(is) |> 
  map(pluck, 1) |> 
  stack() |> 
  filter(values == "element_text")
```

</details>

```
text
title
axis.title
axis.title.x
axis.title.x.top
axis.title.x.bottom
axis.title.y
axis.title.y.left
axis.title.y.right
axis.text
axis.text.x
axis.text.x.top
axis.text.x.bottom
axis.text.y
axis.text.y.left
axis.text.y.right
legend.text
legend.title
plot.title
plot.subtitle
plot.caption
plot.tag
strip.text
strip.text.x
strip.text.y
strip.text.y.left
```

**Comment by karawoo** [comment]
You are correct (thanks for watching the talk 😄 ). For `theme_grey()` it's intentional that the axis text is a different color than the default text color for the theme; you'll want to set the axis text specifically if you want to override it.

**Comment by rjake** [comment]
Thanks, @karawoo. It's very cool how you all set this up. It seems `theme_grey()` sits under all the theme defaults (except `theme_void`). Would it be possible to add a theme that allows everything under `.element_tree` to flow through? 

https://github.com/tidyverse/ggplot2/blob/a979ffd26cdb456d54e2671c2eed16c65bc878b7/R/theme-elements.r#L402-L408

**Comment by karawoo** [comment]
There's no way to do that and have a reasonable looking theme. You'd have to only set the top-level elements and let everything else inherit from those, which means that every text element, line, and rectangle on the entire plot would have the same color, size, orientation, margins, and justification. You'd end up needing to customize everything anyway, even more so than with `theme_grey()`. I'm going to go ahead and close this issue as the theme inheritance system is working as intended.
